src/components/views/login/InteractiveAuthEntryComponents.js

```diff
@@ -209,6 +210,103 @@ export const RecaptchaAuthEntry = React.createClass({
     },
 });
 
+export const TermsAuthEntry = React.createClass({
+    displayName: 'TermsAuthEntry',
+
+    statics: {
+        LOGIN_TYPE: "m.login.terms",
+    },
+
+    propTypes: {
+        submitAuthDict: PropTypes.func.isRequired,
+        stageParams: PropTypes.object.isRequired,
+        errorText: PropTypes.string,
+        busy: PropTypes.bool,
+    },
+
+    componentWillMount: function() {
+        const allPolicies = this.props.stageParams.policies || {};
+        const prefLang = SettingsStore.getValue("language");
+        const initToggles = {};
+        const pickedPolicies = [];
+        for (const policyId of Object.keys(allPolicies)) {
+            const policy = allPolicies[policyId];
+
+            // Pick a language based on the user's language, falling back to english,
+            // and finally to the first language available. If there's still no policy
+            // available then the homeserver isn't respecting the spec.
+            const availableLangs = Object.keys(policy).filter(e => e !== "version");
+            let langPolicy = policy[prefLang];
+            if (!langPolicy) langPolicy = "en";
```

Comment:
Seems like the code below assumes `langPolicy` is an object. Should this be `policy.en`?